pkgs/development/compilers/ocaml/4.12.nix

```
@@ -1,9 +1,9 @@
 import ./generic.nix {
   major_version = "4";
   minor_version = "12";
-  patch_version = "0-alpha1";
+  patch_version = "0-alpha2";
   src = fetchTarball {
-    url = "http://caml.inria.fr/pub/distrib/ocaml-4.12/ocaml-4.12.0~alpha1.tar.xz";
-    sha256 = "1p9nnj7l43b697b6bm767znbf1h0s2lyc1qb8izr1vfpsmnm11ws";
+    url = "http://caml.inria.fr/pub/distrib/ocaml-4.12/ocaml-4.12.0~alpha2.tar.xz";
```

If you make this a recursive set, you could auto-generate the URL, probably.

```suggestion
    url = "http://caml.inria.fr/pub/distrib/ocaml-${major_version}.${minor_version}/ocaml-${major_version}.${minor_version}.${patch_version}.tar.xz";
```